source/_integrations/template.markdown

```diff
@@ -387,109 +491,207 @@ sensor:
 
 {% endraw %}
 
-### Change the Friendly Name Used in the Frontend
+### Change the name
 
-This example shows how to change the `friendly_name` based on a state.
+This example shows how to change the `name` based on a state.
 
 {% raw %}
 
 ```yaml
-sensor:
-  - platform: template
-    sensors:
-      net_power:
-        friendly_name_template: >-
+template:
+  - sensor/:
+      - name: >
           {% if states('sensor.power_consumption')|float < 0 %}
             Power Consumption
           {% else %}
             Power Production
           {% endif %}
-        value_template: "{{ states('sensor.power_consumption') }}"
+        state: "{{ states('sensor.power_consumption') }}"
         unit_of_measurement: "kW"
 ```
 
 {% endraw %}
 
-### Add Custom Attributes
+### Working without entities
 
-This example shows how to add custom attributes.
+The `template` sensors are not limited to use attributes from other entities but can also work with [Home Assistant's template extensions](/docs/configuration/templating/#home-assistant-template-extensions).
+
+This template defines a trigger to update at midnight.
 
 {% raw %}
 
 ```yaml
-sensor:
-  - platform: template
-    sensors:
-      my_device:
-        value_template: >-
-          {% if is_state('device_tracker.my_device_nmap','home') %}
-            Home
-          {% else %}
-            {{ states('device_tracker.my_device_gps') }}
-          {% endif %}
-        attribute_templates:
-          latitude: >-
-            {% if is_state('device_tracker.my_device_nmap','home') %}
-              {{ state_attr('zone.home','latitude') }}
-            {% else %}
-              {{ state_attr('device_tracker.my_device_gps','latitude') }}
-            {% endif %}
-          longitude: >-
-            {% if is_state('device_tracker.my_device_nmap','home') %}
-              {{ state_attr('zone.home','longitude') }}
-            {% else %}
-              {{ state_attr('device_tracker.my_device_gps','longitude') }}
-            {% endif %}
+template:
+  - trigger:
+      - platform: time_pattern
+        # This will update every night
+        hours: 0
+        minutes: 0
+    sensor:
+      - name: Not smoking
+        state: '{{ ( ( as_timestamp(now()) - as_timestamp(strptime("06.07.2018", "%d.%m.%Y")) ) / 86400 ) | round(2) }}'
+        unit_of_measurement: "Days"
 ```
 
 {% endraw %}
 
-### Working without entities
+## Legacy Sensor configuration format
```

Comment:
I'd order the legacy sections (binary sensor and sensor) 🔤.

Reply:
Have moved legacy binary senesor above sensor docs. And swapped  the link at the top to point at binary sensor. That's what you meant right?

Reply:
Yes, exactly.